app/controllers/search_controller.rb

```diff
@@ -5,7 +5,11 @@ def index
     if @form.valid?
       @result = FirmRepository.new.search(@form.to_query, page: page)
     else
-      render 'landing_page/show'
+      if params[:origin] == 'results'
```

Comment:
yeah, sorry - I did that nasty as a quick working concept, def. needs some tidying up!